internal/tiger/cmd/integration_test.go

```diff
@@ -355,11 +355,483 @@ func TestServiceLifecycleIntegration(t *testing.T) {
+		}
+
+		// Verify all options in output
+		if result["role_name"] != roleName {
+			t.Errorf("Expected role_name=%s, got: %v", roleName, result["role_name"])
+		}
+		if readOnly, ok := result["read_only"].(bool); !ok || !readOnly {
+			t.Errorf("Expected read_only=true, got: %v", result["read_only"])
+		}
+		// Note: statement_timeout not checked as it cannot be set with --from tsdbadmin
+
+		// Verify from_roles in output
+		fromRoles, ok := result["from_roles"].([]interface{})
+		if !ok || len(fromRoles) == 0 {
+			t.Error("Expected from_roles in output")
+		} else if fromRoles[0] != "tsdbadmin" {
+			t.Errorf("Expected from_roles to contain 'tsdbadmin', got: %v", fromRoles)
+		}
+
+		t.Logf("✅ Successfully created role with all valid options: %s", roleName)
+	})
+
+	t.Run("CreateRole_VerifyRolesExist", func(t *testing.T) {
+		if serviceID == "" {
+			t.Skip("No service ID available from create test")
+		}
+
+		if len(createdRoles) == 0 {
+			t.Skip("No roles created to verify")
+		}
+
+		// Verify all created roles exist by querying pg_roles
+		t.Logf("Verifying created roles exist in database")
+
+		for _, roleName := range createdRoles {
+			output, err := executeIntegrationCommand(
+				"db", "psql", serviceID,
+				"--", "-c", fmt.Sprintf("SELECT rolname FROM pg_roles WHERE rolname = '%s';", roleName),
+			)
+
+			if err != nil {
+				t.Errorf("Failed to verify role %s exists: %v\nOutput: %s", roleName, err, output)
+				continue
+			}
+
+			// Verify role name appears in output
+			if !strings.Contains(output, roleName) {
+				t.Errorf("Role %s not found in pg_roles query output: %s", roleName, output)
+			} else {
+				t.Logf("✅ Verified role exists in database: %s", roleName)
+			}
+		}
+	})
+
+	t.Run("CreateRole_DuplicateName", func(t *testing.T) {
+		if serviceID == "" {
+			t.Skip("No service ID available from create test")
+		}
+
+		if len(createdRoles) == 0 {
+			t.Skip("No roles created to test duplicate")
+		}
+
+		// Try to create a role with the same name as the first created role
+		duplicateRoleName := createdRoles[0]
+
+		t.Logf("Attempting to create duplicate role: %s", duplicateRoleName)
+
+		output, err := executeIntegrationCommand(
+			"db", "create", "role", serviceID,
+			"--name", duplicateRoleName,
+		)
+
+		// This should fail with a role already exists error
+		if err == nil {
+			t.Errorf("Expected duplicate role creation to fail, but got output: %s", output)
+		} else {
+			// Verify error message indicates role already exists
+			if !strings.Contains(err.Error(), "already exists") && !strings.Contains(output, "already exists") {
+				t.Logf("Note: Error message may not contain 'already exists': %v\nOutput: %s", err, output)
+			} else {
+				t.Logf("✅ Duplicate role creation correctly failed")
+			}
+		}
+	})
+
 	t.Run("DeleteService", func(t *testing.T) {
 		if serviceID == "" {
 			t.Skip("No service ID available for deletion")
 		}
 
+		// Clean up roles before deleting the service
+		cleanupRoles()
+
```

Comment:
Is it really necessary to drop the roles right before we delete the service? Isn't deleting the service enough? Could probably get rid of the `cleanupRoles()` function entirely, I think (since this is the only place it's called).